src/classes/OpportunityTriggerHelper.cls

```diff
@@ -0,0 +1,421 @@
+                                        else {
+                                            newRecord.Name = recordName;
+                                        }
+                                        newRecord.Account__c = opp.AccountId;
+                                        newRecord.Opportunity__c = opp.Id;
+                                        createNewAutomobileRecords.add(newRecord);
+                                    }
+                                }
+                                else {
+                                    opp.addError('Invalid Record : ' + recordName);
+                                }
+                            }
+                            else {
+                                opp.addError('Duplicate not allowed' + recordName);
+                            }
+                        }
+                    }
+                }
+                else {
+                    opp.AccountId.addError('Installation Account is Required');
+                }
+            }
+        }
+
+        Boolean flag = true;
+        if (!createNewAutomobileRecords.isEmpty()) {
+            // Insert New Automobile Records
+            flag = true;
+            sendMail(createNewAutomobileRecords, updateExistingAutomobileRecors, listOfOpportuinties, flag);
+        }
+        if (!updateExistingAutomobileRecors.isEmpty()) {
+            // Update Existing Automobile Records
+            flag = false;
+            sendMail(createNewAutomobileRecords, updateExistingAutomobileRecors, listOfOpportuinties, flag);
+        }
+    }
+
+    /*
+    ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
+    * Created Date         :       05/08/2022
+    * Author               :       Bharat Kumar
+    * Description          :       Send Mail when Opportunity is updated or created
+    * Returns              :       Void
+    * Parameters           :       List<SObject>, List<SObject>, Set<Opportunity>, Boolean
+    ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+    */
+
+    public static void sendMail(List<SObject> listOfNewRecords, List<SObject> listOfUpdatedRecords, Set<Opportunity> setOfOpportuinties, Boolean flag) {
+
+        Map<String, SObject> mapOfRecordsWithOpportunity = new Map<String, Opportunity>();
+        Map<String, List<SObject>> mapOfAutomobileRecords = new Map<String, List<SObject>>();
+        List<SObject> listOfAutomobileRecords = new List<SObject>();
+
+        if (!listOfNewRecords.isEmpty() && !listOfUpdatedRecords.isEmpty()) {
+            listOfAutomobileRecords.addAll(listOfNewRecords);
+            listOfAutomobileRecords.addAll(listOfUpdatedRecords);
+        }
+        else if (!listOfNewRecords.isEmpty()) {
```

Comment:
remove the 243 if statement, instead use Individual If for size check on both lists

Reply:
Done